src/portal/mod.rs

```diff
@@ -217,18 +236,26 @@ impl Keyring {
     }
 }
 
+/// Return key, derive and store it first if not initialized
+async fn derive_key<'a>(
+    key: &'a mut OnceCell<crate::Key>,
+    keyring: &api::Keyring,
+    secret: &[u8],
```

Comment:
```suggestion
    secret: &Secret,
```
is more accurate no?

Reply:
changed